"<:div><script>alert()<:script>{{1+1}}.java

```diff
@@ -0,0 +1,68 @@
+package cucumber.helpers.shared;
+
+import com.amazonaws.services.dynamodbv2.LockItem;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+public class PackageLocksHelper {
+    private static final Logger log = LogManager.getLogger(PackageLocksHelper.class);
+    private final PackageLock packageLock;
+    private final String stage;
+    private final Map<String, LockItem> lockedPackages = new ConcurrentHashMap<>();
+
+    public PackageLocksHelper(PackageLock packageLock, String stage) {
+        this.packageLock = packageLock;
+        this.stage = stage.substring(0, 1).toUpperCase() + stage.substring(1).toLowerCase();
+    }
+
+    public String lockGitFarmPackageWithStage(String packagePrefix) {
+        int count = this.packageLock.getCountForPackage(packagePrefix);
+        List<String> packagePool = IntStream.rangeClosed(1, count)
+                .mapToObj(String::valueOf)
+                .map(i -> packagePrefix + "-" + stage + i)
+                .collect(Collectors.toList());
+        return this.lockGitFarmPackage(packagePool);
+    }
+
+    public String lockGitFarmPackage(String packagePrefix) {
+        int count = this.packageLock.getCountForPackage(packagePrefix);
+        List<String> packagePool = IntStream.rangeClosed(1, count)
+                .mapToObj(String::valueOf)
+                .map(i -> packagePrefix + i)
+                .collect(Collectors.toList());
+        return this.lockGitFarmPackage(packagePool);
+    }
+
+    public String lockGitFarmPackage(List<String> packagePool) {
+        try {
+            LockItem lockPackage = this.packageLock.getLockedPackage(packagePool);
+            String packageName = lockPackage.getSortKey().get();
+
+            this.lockedPackages.put(packageName, lockPackage);
+            log.info("A lock is acquired for GitFarm package " + packageName);
+            return packageName;
+        } catch (InterruptedException e) {
+            throw new RuntimeException(e);
+        }
+    }
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

By convention, any method that exits by throwing an InterruptedException clears the thread’s interrupt status. Consider setting the flag using Thread.currentThread().interrupt() so that higher-level interrupt handlers can appropriately function. Only code that implements a thread\u0027s interruption policy may swallow an interruption request. For more information, see [https://docs.oracle.com/javase/tutorial/essential/concurrency/interrupt.html](https://docs.oracle.com/javase/tutorial/essential/concurrency/interrupt.html).